x/session/keeper/query_get_session_test.go

```diff
@@ -91,34 +93,69 @@ func TestSession_GetSession_Failure(t *testing.T) {
 		{
 			name: "application address does not reflected a staked application",
 
-			appAddr:     "some string that is not a valid app address",
+			appAddr:     sample.AccAddress(), // a random (valid) app address that's not staked
 			serviceId:   keepertest.TestServiceId1,
 			blockHeight: 1,
 
 			expectedErrContains: types.ErrAppNotFound.Error(),
 		},
 		{
-			name: "service ID does not reflect one with staked suppliers",
+			name: "application staked for service that has no available suppliers",
 
 			appAddr:     keepertest.TestApp1Address,
-			serviceId:   "some string that is not a valid service Id",
+			serviceId:   keepertest.TestServiceId11,
 			blockHeight: 1,
 
 			expectedErrContains: types.ErrSuppliersNotFound.Error(),
 		},
+		{
+			name: "application is valid but not staked for the specified service",
+
+			appAddr:     keepertest.TestApp1Address,
+			serviceId:   "svc9001", // App1 is not staked for service over 9000
+			blockHeight: 1,
+
+			expectedErrContains: types.ErrAppNotStakedForService.Error(),
+		},
+		{
+			name: "application address is invalid format",
+
+			appAddr:     "invalid_app_address",
+			serviceId:   keepertest.TestServiceId1,
+			blockHeight: 1,
+
+			expectedErrContains: types.ErrSessionInvalidAppAddress.Error(),
+		},
+		{
+			name: "service ID is invalid",
+
+			appAddr:     keepertest.TestApp1Address,
+			serviceId:   "invalid_serviceId",
+			blockHeight: 1,
+
+			expectedErrContains: "invalid serviceID for session being retrieved",
```

Comment:
Not sure if it's relevant and/or there's any action item  but I just  wanted to call out that I think this might also be invalid because it's too long. IIRC, I saw somewhere that the max length for a serviceId was 7 or 8.

Reply:
Updated the serviceId to `serviceId:   "service_id_is_too_long_to_be_valid",`